IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+
+#import <Foundation/Foundation.h>
+#import "MSIDCredentialCacheItem.h"
+#import "MSIDDefaultCredentialCacheKey.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSIDMacCredentialStorageItem : NSObject <MSIDJsonSerializable>
```

Comment:
This class has similar logic with MSIDCache. Why don't we subclass or aggregate it?

Reply:
Even if we subclass it , we will need to rewrite most of the implementations of the subclass like setObject objectForKey because of the way MSIDDefaultCredentialKey is passed and then further converted to NSString. Also, the copy method will need to change to deep copy as it is a nested dictionary. Another option would be to refactor the code to pass nsstring as key. I would prefer to do that in a separate PR if its not that urgent.

Reply:
aggregation is a good idea. I'm fine taking it as a follow up though to unblock this PR (with 3 other follow ups :))